node/src/components/consensus/protocols/highway/tests.rs

```diff
@@ -151,6 +151,68 @@ fn send_a_wire_unit_with_too_small_a_round_exp() {
     }
 }
 
+#[test]
+fn send_a_valid_wire_unit() {
+    let creator: ValidatorIndex = ValidatorIndex(0);
+    let state: State<ClContext> = new_test_state(&[100.into()], 0);
+    let panorama: ValidatorMap<Observation<ClContext>> = Panorama::from(vec![N, N]);
+    let seq_number = panorama.next_seq_num(&state, creator);
+    let mut rng = TestRng::new();
+    let wunit: WireUnit<ClContext> = WireUnit {
+        panorama,
+        creator,
+        instance_id: ClContext::hash(INSTANCE_ID_DATA),
+        value: Some(CandidateBlock::new(
+            ProtoBlock::new(vec![], vec![], false),
+            vec![],
+        )),
+        seq_number,
+        timestamp: 0.into(),
+        round_exp: 14,
+        endorsed: BTreeSet::new(),
+    };
+    let alice_keypair: Keypair = Keypair::from(Rc::new(ALICE_SECRET_KEY.clone()));
+    let highway_message: HighwayMessage<ClContext> = HighwayMessage::NewVertex(Vertex::Unit(
+        SignedWireUnit::new(wunit, &alice_keypair, &mut rng),
+    ));
+    let mut highway_protocol = new_test_highway_protocol(vec![]);
+    let sender = NodeId(123);
+    let msg = bincode::serialize(&highway_message).unwrap();
+    let mut effects = highway_protocol.handle_message(sender, msg, false, &mut rng);
+    assert_eq!(effects.len(), 1);
+
+    let opt_protocol_outcome = effects.pop();
+
+    let mut new_effects = match &opt_protocol_outcome {
+        None => panic!("We just checked that effects has length 1!"),
+        Some(ProtocolOutcome::ValidateConsensusValue(node_id, candidate_block, timestamp)) => {
+            assert_eq!(*node_id, NodeId(123));
+            assert_eq!(*timestamp, Timestamp::from(0));
+            highway_protocol.resolve_validity(candidate_block, true, &mut rng)
```

Comment:
Once https://github.com/CasperLabs/casper-node/pull/653 merges, you won't need to resolve the validity if you propose an empty vector of deploys.

Reply:
Just merged.

Reply:
I merged the new `master`, but `ValidateConsensusValue` is still being produced in the test :thinking:

Reply:
I approved, although I'm really confused about this, and it would be good to find out why before merging…

Reply:
I know, I changed from `is_empty` to `needs_validation` but didn't change the implementation :facepalm: it should be a negation now! `is_empty` to `!is_empty`.

Reply:
It might be because the logic is reversed... :grimacing: https://github.com/CasperLabs/casper-node/blob/master/node/src/components/consensus/candidate_block.rs#L44-L46

Reply:
I'll fix this. And we'll have a test that checks this! :P

Reply:
Can you change (inverse) all the places where `needs_validation` is implemented?

Reply:
Sorry, I completely missed that in the review. :man_facepalming: 
(My bad: I was the one suggesting to replace the method with its negation…)